File: 0448-find-all-numbers-disappeared-in-an-array/0448-find-all-numbers-disappeared-in-an-array.cpp
class Solution {
public:
    vector<int> findDisappearedNumbers(vector<int>& nums) {
        
        	map<int , int>mp;
	int n = (int)nums.size();

	for(int i = 0 ; i < n ; i ++){
		mp[nums[i]] ++;
	}

	vector<int> ans;
	for(int i = 1 ; i <= n ; i ++){
		if(mp[i] > 0)continue;
		ans.push_back(i);
	}

	return ans;
        
        
    }
};

File: 0448-find-all-numbers-disappeared-in-an-array/0448-find-all-numbers-disappeared-in-an-array.py
class Solution:
    def findDisappearedNumbers(self, nums: List[int]) -> List[int]:
            ans = []

            for n in nums:

                i = abs(n) - 1
                nums[i] = -1 * abs(nums[i])


            for i in range(len(nums)):
                if nums[i] > 0:
                    ans.append(i + 1)


            return ans
